### PR TITLE
Feature/test migration

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes

### CODE REVIEW

File: inc/dbmysql.class.php [hunk 1]
```diff
@@ -714,4 +714,52 @@ static public function checkForCrashedTables() {
       }
       return $crashed_tables;
    }
+
+   /**
+   * Verify if a DB table exists
+   *
+   * @param string $tablename Name of the table we want to verify.
+   *
+   *  @return boolean
```
Comment (orthagh) [inline inc/dbmysql.class.php]: double space here

File: inc/dbmysql.class.php [hunk 2]
```diff
@@ -714,4 +714,52 @@ static public function checkForCrashedTables() {
...
+      $this->free_result($result);
+      return false;
+   }
+
+   /**
+   * Verify if a DB field exists
+   *
+   * @param string  $table    Name of the table we want to verify.
+   * @param string  $field    Name of the field we want to verify.
+   * @param Boolean $usecache If use field list cache (default true)
```
Comment (orthagh) [inline inc/dbmysql.class.php]: i think it's a copypasta of the old method (so not your fault) but this doc line is pretty abstruse

File: inc/migration.class.php [hunk 3]
```diff
@@ -910,20 +913,23 @@ public function addConfig($values) {
...
-      foreach ($existing as $conf) {
-         unset($this->configs[$conf['name']]);
-      }
       if (count($this->configs)) {
-         Config::setConfigurationValues($this->context, $this->configs);
-         $this->displayMessage(sprintf(
-            __('Configuration values added for %1$s.'),
-            implode(', ', $this->configs)
-         ));
+         $existing = $DB->request(
```
Comment (orthagh) [inline inc/migration.class.php]: why this change ?

Comment (trasher) [inline inc/migration.class.php]: To use dbiterator
